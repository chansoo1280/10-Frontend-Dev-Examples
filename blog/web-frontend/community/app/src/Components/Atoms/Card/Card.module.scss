@import "@Styles/scss/default.scss";

.card-wrap {
    padding: 54px 24px;
}

.card {
    border-radius: 2px;
}