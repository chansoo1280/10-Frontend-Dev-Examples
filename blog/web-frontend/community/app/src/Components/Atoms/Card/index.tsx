// #region Global Imports
import React from "react"
import classNames from "classnames"
// #endregion Global Imports

// #region Local Imports
import { Space, defaultProps } from "@Components"
import styles from "./Card.module.scss"
import { SpaceProps } from "../Space"
// #endregion Local Imports

export interface CardProps extends defaultProps {
    padding?: string | number
    onClick?: React.MouseEventHandler
    BoxProps?: SpaceProps
}
const Card = (props: CardProps): JSX.Element => {
    const { show, className, padding, children, BoxProps, ...rest } = props
    const prefixCls = "card"
    const classes = classNames(
        styles[`${prefixCls}`],
        {
            [styles[`${prefixCls}--hide`]]: show === false,
        },
        className,
    )
    return (
        <Space className={classes} style={{ padding }} {...BoxProps} {...rest}>
            {children}
        </Space>
    )
}
const Wrap = (props: defaultProps) => {
    const { ...rest } = props
    return <div className={classNames(styles[`card-wrap`])} {...rest} />
}
Card.wrap = Wrap
export default Card
