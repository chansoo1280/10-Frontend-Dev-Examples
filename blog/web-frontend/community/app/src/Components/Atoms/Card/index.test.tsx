import { render, screen } from "@testing-library/react"
import Card from "./"
describe("Card", () => {
    it("renders card unchanged", () => {
        const { container } = render(
            <Card>
                <div>1</div>
            </Card>,
        )
        expect(container).toMatchSnapshot()
    })
    it("should render a card with the class", () => {
        render(
            <Card data-testid="card">
                <div>1</div>
            </Card>,
        )

        const card = screen.getByTestId("card")
        expect(card).toBeInTheDocument()
        expect(card).toHaveClass("card")
    })
})
