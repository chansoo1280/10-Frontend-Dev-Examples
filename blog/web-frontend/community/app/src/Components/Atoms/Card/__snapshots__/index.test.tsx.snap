// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders card unchanged 1`] = `
<div>
  <div
    class="space card"
  >
    <div>
      1
    </div>
  </div>
</div>
`;
